lib/ueberauth/strategy/slack.ex

```diff
@@ -133,23 +134,32 @@ defmodule Ueberauth.Strategy.Slack do
     user = conn.private[:slack_user]
     auth = conn.private[:slack_auth]
     identity = conn.private[:slack_identity]
-    image_urls = (get_in(user, ["profile"]) || %{})
+    profile = (get_in(user, ["profile"]) || get_in(identity, ["user"]) || %{})
+    image_urls = profile
     |> Map.keys
     |> Enum.filter(&(&1 =~ ~r/^image_/))
-    |> Enum.map(&({&1, user["profile"][&1]}))
+    |> Enum.map(&({&1, profile[&1]}))
+    |> Enum.into(%{})
```

Comment:
@ibash `Enum.into/3` also exists and it can be used to do `Enum.map/2` + `Enum.into/2` in one go:

```elixir
|> Enum.into(%{}, &({&1, profile[&1]}))
```

@ueberauth/developers how do you feel about starting to clean up some of these `map/2` + `into/2` calls?